streams/src/test/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreatorTest.java

```diff
@@ -141,82 +255,191 @@ public void shouldFailForUnknownTaskOnStreamsProducerPerTask() {
         {
```

Comment:
Out of curiosity, why do we add two `{}` around the code logic here？

Reply:
I want to make variable `thrown` "final" and also not introduce `thrown2` -- using two blocks allow to use the same name twice and to declare the variable "final". It also make it clear that it's basically two independent tests (we could also separate them into two test methods, but we test the same function and thus it seems better to do both cases in one test).